provider/marathon/marathon.go

```diff
@@ -284,74 +383,86 @@ func (p *Provider) getPort(task marathon.Task, application marathon.Application)
 	return strconv.Itoa(port)
 }
 
-func (p *Provider) getWeight(application marathon.Application) string {
-	if label, ok := p.getLabel(application, types.LabelWeight); ok {
+// Extract weight from labels for a given service and a given marathon application
```

Comment:
Please drop the GoDoc added here and for the other label-based `get*` methods that didn't have a GoDoc before unless it's not immediately clear what they do.